conf/gulp-tasks/template/createcleanfilestask.js

```diff
@@ -0,0 +1,32 @@
+const { getPrecompiledFileName, addLocalePrefix } = require('./filenameutils');
+
+const del = require('del');
+
+/**
+ * Creates a cleanFiles task for the specified locale, which will
+ * clear away any intermediary files in the template build chain.
+ * Also customizes the task's name for display in the command line.
+ * @param {string} locale
+ * @returns {Function}
+ */
+function createCleanFilesTask (locale) {
+  const cleanFilesTask = callback => _cleanFiles(callback, locale);
+  const taskName = addLocalePrefix('cleanFiles', locale);
+  Object.defineProperty(cleanFilesTask, 'name', {
+    value: taskName
+  });
+  return cleanFilesTask;
+}
+
+/**
+ * The cleanFiles task for a specified locale.
+ * @param {Function} callback
+ * @param {string} locale
+ */
+function _cleanFiles (callback, locale) {
```

Comment:
I'm not super familiar with gulp/async. For my own edification, why do we need all these `callback` parameters now?

Reply:
I think what's happening is that we had to change our gulp tasks to be async, to handle initializing i18next. Previously, these tasks would return streams, and gulp already knows how to handle async completion for a stream, so we didn't have to add these callback function.

Now, these tasks return promises that resolve to streams. My guess is that if gulp sees that your tasks returns a promise, it will mark the task as finished as soon as the promise resolves. I don't believe (based purely off experimentation for the past few days) that gulp has logic that says if your task returns a promise BUT it's also a stream, then wait for the stream to finish.

So in order to only have the promise the task returns resolve AFTER the returned stream is finished, I added these callbacks, and also wrapped the returned promise to only resolve after the stream is finished. Gulp was already passing these callbacks as the first argument to every task, we just weren't using them.